apps/data-types.md

```diff
@@ -4,111 +4,145 @@ key: apps-data-types
+| Attribute | Type | Description |
+| - | :-: |  - |
+| name | string | The name of the selected product attribute, e.g. colour. |
+| value | string | The assigned value of the selected product attribute, e.g. white. |
 
-## Example
+# customAttribute
 
-{% highlight text %}
-{"page": 1}
-{% endhighlight %}
+| Attribute | Type | Description |
+| - | :-: |  - |
+| key | string | The identifier of the custom attribute. |
+| displayKey | string | The displayed name of the custom attribute. |
+| singleValue | boolean | `true` if just one feature is selected for the custom attribute.  |
+| type | string | The data type of the custom attribute. |
+| values | array of [value](page:apps-data-types#value)| The options selected for the custom attribute. |
 
-# String
+# image
 
-A sequence of characters.
+This object is used for the attributes of images.
 
-## Example
+| Attribute | Type | Description |
+| - | :-: |  - |
+| url | string | The URL of an image. |
+| classifier | string | Specifies the image, e.g. Thumbnail or Medium. |
 
-{% highlight text %}
-{"shopId": "CreamyIceShop"}
-{% endhighlight %}
 
-# Enumeration
+# lineItemContainer
 
-Consists a set of named values.
+| Attribute | Type | Description |
+| - | :-: |  - |
+| grandTotal | [Price](page:apps-data-types#price) | The total price including product price, shipping and tax. |
+| totalBeforeTax | [Price](page:apps-data-types#price) | The total price including product price, shipping excluding tax. |
+| totalTax | [Price](page:apps-data-types#price) | The total amount of the tax. |
+| lineItemsSubTotal | [Price](page:apps-data-types#price) | The sum of the line item price of all line items. |
+| productLineItems | Array of [productLineItem](page:apps-data-types#productlineitem) | A list of line items. |
 
-## Example
+# link
 
-{% highlight text %}
-{"enum": ["oldest", "newest"]}
-{% endhighlight %}
+This object is used for the attributes of links.
 
-# Boolean
+| Attribute | Type | Description |
+| - | :-: |  - |
+| rel | string | The link relation that describes how the link relates to the call. |
+| href | string | The URL of the related link that can be used for subsequent calls. |
+| title | string | The title of the item that is linked. (optional)  |
 
-True or false values.
+# price
 
-## Example
+This object is used for the attributes of basePrice, depositPrice, ecoParticipationPrice, manufacturerPrice, grandTotal, priceWithDeposits, totalBeforeTax, totalTax and lineItemsSubTotal.
 
-{% highlight text %}
-{"layout": "page", "title": "use case", "searchable": true}
-{% endhighlight %}
+| Attribute | Type | Description |
+| - | :-: |  - |
+| taxType | string | Indicates if the amount includes tax, e.g. gross. |
+| formatted | string | The amount of the price with currency unit. |
+| amount | number | The amount of the price. |
+| currency | string | The currency code of the price according to ISO 4217. |
 
-# Array
+# priceInfo
```

Comment:
add to priceInfo:
- depositPrice (Price)
- ecoParticipationPrice (Price)
- manufacturerPrice (Price)
- priceWithDeposits (Price)
- basePrice (BasePrice)

where BasePrice is an object with the following attributes:
refQuantity (Quantity)
refPrice (Price)
formatted (String)
quantity (Quantity)